src/compositemark/callout.ts

```diff
@@ -1,30 +1,25 @@
 import {Config} from '../config';
 import {Encoding} from '../encoding';
 import * as log from '../log';
-import {MarkConfig, MarkDef} from '../mark';
+import {MarkConfig} from '../mark';
 import {GenericUnitSpec, LayerSpec} from '../spec';
 
-
 export const CALLOUT: 'callout' = 'callout';
 export type CALLOUT = typeof CALLOUT;
 
-export interface CalloutDef {
+export interface CalloutDef extends CalloutConfig {
   type: CALLOUT;
-  calloutAngle?: number;
-  calloutOffset?: number;
-  calloutLength?: number;
-  labelOffset?: number;
-  callout?: MarkDef;
-  label?: MarkDef;
+  callout?: MarkConfig;
```

Comment:
callout mark is already the whole thing so I don't think we should call the line segment part a `callout`.   After all, this should be consistent with config, and `config.callout.callout` is confusing.

Reply:
I wonder if it makes sense to support `style` and `clip` too, but definitely not type?

(Maybe not -- I'm just asking)

Reply:
These two (`label` and whatever you gonna call `rule/line/callout`) should be in CalloutConfig anyway.